naming/README.md

```diff
@@ -0,0 +1,93 @@
+# Solana Naming Service [WIP]
+
+A spl program for issuing and managing ownership of: domain names, Solana Pubkeys, URLs, twitter handles, ipfs cid's etc..
+
+This program can be used for dns, pubkey etc lookups via a browser extension (wip), the goal is to create an easy way to identify Solana public keys with various links.
+
+Definitions: Names are strings that map to records which can hold multiple values.\
+ValueType defines the type (url, uri, solkey, etc..) of the value stored following the specification of the js bindings that will read from the name accounts.
+
+- For each name it's values will be stored in the data of a program account. The address of this account is derived from the name.
+In this account the pubkey of the owner (who can change the values) is also stored.
+
+- Domain name declarations need to be verified by the owner associated to the parent domain. That owner can also be set to Null, removing the permissions needed for a specific domain name and its sub-domains. The declaration of a new TLD (top-level domain) name needs to be verified by the root TLD owner as it is a sub-domain of it. Therefore the owner of the root domain can administrate the TLD and sub-domain distribution.
+Similarly to DNS, we use `"."` as a root domain name seed. Nevertheless, as we need to protect the domain names from collisions with DNS names, the root tld is written out as the string `".sol"` for url lookups in the browser search bar via the extension.
+For example the name resolution of `something.sol` would use the seeds: `[["something"], ["sol"], ["."]]` (converting the strings to bytes of course) but would be needed to be written as `something.sol.sol` in the browser bar.
+
+- Twitter handles can be added as sub-domain names of one TLD name the owner of which will therefore hold the right to add a twitter handle name. This enables the verification of twitter accounts for example by asking the user to tweet his pubkey or a signed message. A bot that holds the private twitter-domain-name owner key can then sign the AddName instruction and send it back to the user who will then submit it to the program. This verification bot acts as a kind of centralized oracle that verifies the names. For now a centralized set up seems appropriate for this. Additional js bindings that structure this will be implemented.
+
+- [Proposal] An additional instruction (or another contract) could do the following: mint an NFT for an already existing name and activate ownership for the holder of the NFT only. This would enable easy trading of names.
+
+
+Program Interface:
+
+```
+pub enum NameStatus {
+    Deleted,
+    Active,
+}
+
+pub type ValueType = u16;
+
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct NameState {
+    status: u8,
+    values: Vec<(ValueType, String)>,
```

Comment:
Here's a question - is there a constraint on having multiple of the same ValueType in this vec? Like what if there are two SOL addresses in here, or two twitter handles? Is it up to the client reading it to interpret that how they will?

Reply:
Ah yes I wanted to keep this limitation as otherwise you open up to a whole lot of prioritization problems between two values indeed. But if we agree on a natural order of interpretation (maybe just the order in the Vec?) then it sounds feasible.